### PR TITLE
dev-marie-webpage: Zeolite QA and visualisation

### PR DESCRIPTION
**Pull request by picas9dan**

### CODE REVIEW

**Review by sm453 — CHANGES_REQUESTED** [review]
As before, the most glaring issue is the total absence of any documentation. Given that this PR adds substantial new functionality, can you please substantially (i.e. in a way that reflects the expansion in functionality) expand the main readme file(s)?

File: QuestionAnswering/QA_SEQ2SEQ/deployment/triton/model_repository/seq2seq_chemistry/1/model.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-    "model_path": "picas9dan/20231130_6_onnx_8bit"
+    "model_path": "picas9dan/20240516_8_onnx_8bit"
```
[the remark this answers is not in the document]
Comment (picas9dan) [reply QuestionAnswering/QA_SEQ2SEQ/deployment/triton/model_repository/seq2seq_chemistry/1/model.json:2]: By default, the code will interpret the `model_path` argument as the identifier on HuggingFace Hub, which is [here](https://huggingface.co/picas9dan/20240516_8_onnx_8bit). Currently the code is not set up to load model weights from local filesystem. 

I'll update the README to reflect this information.

[the remark this answers is not in the document]
Comment (picas9dan) [reply QuestionAnswering/QA_SEQ2SEQ/deployment/triton/model_repository/seq2seq_chemistry/1/model.json:2]: I've updated the README file at `deployment/triton` to reflect what model type is supported by the code, how to convert the model weights to the required format, and the platform to host the weights.

File: QuestionAnswering/QA_SEQ2SEQ/deployment/fastapi_app/api/adv_search.py [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+
+    for i in range(1, 4):  # up to 3 elements
+        attr_guest = f"guest{i}"
+        guest_id = getattr(params, attr_guest)
+        if guest_id:
+            iri = query_species(guest_id)
+            base_query += f"""
+            ?zeo_material zeo:hasGuestCompound ?guest{i}IRI .
+            FILTER(?guest{i}IRI = <{iri}>)
+            SERVICE <http://178.128.105.213:3838/blazegraph/namespace/ontospecies/sparql> {{
```
[the remark this answers is not in the document]
Comment (picas9dan) [reply QuestionAnswering/QA_SEQ2SEQ/deployment/fastapi_app/api/adv_search.py]: I've replaced this with the value of an environment variable.

**Comment by picas9dan** [comment]
> As before, the most glaring issue is the total absence of any documentation. Given that this PR adds substantial new functionality, can you please substantially (i.e. in a way that reflects the expansion in functionality) expand the main readme file(s)?

I have updated `deployment/fastapi_app/README.md` with a description of all the features available.

**Review by sm453 — APPROVED** [review]

**Pull request merged**